### PR TITLE
INT-4398: Replace Gemfire dependency with Geode

### PR DESCRIPTION
**Pull request by artembilan**
JIRA: https://jira.spring.io/browse/INT-4398

* Polishing `gemfire.adoc` and mention the procedure for an
exclusion/inclusion appropriate dependency

### CODE REVIEW

**Comment by artembilan** [comment]
@garyrussell , @jxblum,

Let me know if this simple solution is appropriate!

Or do we definitely don't have choice and have to have two separate projects on the matter ?

Thanks

File: src/reference/asciidoc/gemfire.adoc [hunk 1]
```diff
@@ -1,16 +1,38 @@
...
+GemFire is a distributed data management platform providing a key-value data grid along with advanced distributed system features such as event processing, continuous querying, and remote function execution.
+This guide assumes some familiarity with commercial https://pivotal.io/pivotal-gemfire[Pivotal GemFire] or Open Source http://geode.apache.org[Apache Geode].
 
-Spring integration provides support for GemFire by providing inbound adapters for entry and continuous query events, an outbound adapter to write entries to the cache, and `MessageStore` and `MessageGroupStore` implementations.
-Spring integration leverages thehttp://www.springsource.org/spring-gemfire[Spring Gemfire] project, providing a thin wrapper over its components.
+Spring integration provides support for GemFire implementing inbound adapters for entry and continuous query events, an outbound adapter to write entries to the cache, and `MessageStore` and `MessageGroupStore` implementations.
+Spring integration leverages the http://projects.spring.io/spring-data-gemfire[Spring Data Gemfire] project, providing a thin wrapper over its components.
```
Comment (garyrussell) [inline src/reference/asciidoc/gemfire.adoc]: Also `GemfireLockRegistry`,

**Comment by garyrussell** [comment]
How will that work? You previously committed import changes, e.g. in `GemfireLockRegistry`...

```java
import org.apache.geode.cache.Cache;
import org.apache.geode.cache.Region;
import org.apache.geode.cache.Scope;
```

**Comment by artembilan** [comment]
Pivotal GemFire now is aligned with Apache Geode packaging. That's exactly what John means in his blog post: https://spring.io/blog/2017/10/26/diff-q-spring-data-gemfire-spring-data-geode

**Comment by garyrussell** [comment]
Interesting; a link to that post would have been useful in this PR (and we have it now 😄 ).

**Comment by jxblum** [comment]
Hi @artembilan -

The changes look good.  2 things...

1) The following sentence (line 16) in gemfie.adoc...

> _To switch to commercial Pivotal GemFire based Spring Data Gemfire, there is need to *exclude* `spring-data-geode` from dependencies and add `spring-data-gemfire`_

Might read better as ...

"To switch to commercial Pivotal GemFire based Spring Data GemFire, exclude `spring-data-geode` from the dependencies and add `spring-data-gemfire`."

2) Also, as an FYI, I have been asked by @dussab and the Pivotal Legal team to switch the branding on Spring Data GemFire.  It will change to _Spring Data for Pivotal GemFire_, which is consistent with , and matches the branding for _Spring Data for Apache Geode_ (which not unlike what Apache required us to do with other Apache based _Spring_ projects, e.g. _Spring Data for Apache Cassandra_, _Spring for Apache Hadoop_, etc).  I have filed [SGF-732](https://jira.spring.io/browse/SGF-732).

If you have another other questions, let me know.

Cheers,
John

**Review by garyrussell — COMMENTED** [review]
That's all; we can then merge.

**Pull request merged**